plugins/home/backstage-plugin-home-markdown/src/apis/index.ts

```diff
@@ -0,0 +1,131 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createApiRef, OAuthApi } from '@backstage/core-plugin-api';
+import { Octokit } from '@octokit/rest';
+import { BASE_URL } from '../MarkdownCard/types';
+
+export type GetContentResponse = {
+  content: string;
+  media: Record<string, string>;
+  links: Record<string, string>;
+};
+
+export type GetContentProps = {
+  owner: string;
+  repo: string;
+  branch?: string;
+  path: string;
+};
+
+export interface GithubApi {
+  getContent(props: GetContentProps): Promise<GetContentResponse>;
+}
+
+const mimeTypeMap: Record<string, string> = {
+  svg: 'image/svg+xml',
+  png: 'image/png',
+  jpg: 'image/jpeg',
+  jpeg: 'image/jpeg',
+  gif: 'image/gif',
+  webp: 'image/webp',
+};
+
+const mimeTypeLookup = (href: string): string | undefined => {
+  const match = href.match(/\.(\S+)$/);
+  const extension = match ? match[1] : undefined;
+  return extension ? mimeTypeMap[extension.toLowerCase()] : undefined;
+};
+
+export const githubApiRef = createApiRef<GithubApi>({
+  id: 'plugin.github.service',
+});
+
+const getRepositoryDefaultBranch = (url: string) => {
+  return new URL(url).searchParams.get('ref');
+};
+
+export class GithubClient implements GithubApi {
+  private githubAuthApi: OAuthApi;
+
+  constructor(deps: { githubAuthApi: OAuthApi }) {
+    this.githubAuthApi = deps.githubAuthApi;
+  }
+
+  async getContent(props: {
+    owner: string;
+    repo: string;
+    branch?: string;
+    path: string;
+  }): Promise<{
+    content: string;
+    media: Record<string, string>;
+    links: Record<string, string>;
+  }> {
+    const { path, repo, owner, branch } = props;
+    const token = await this.githubAuthApi.getAccessToken();
+    const octokit = new Octokit({ auth: token });
+    const response = await octokit.request(
+      `GET /repos/${owner}/${repo}/contents/${path}`,
+      {
+        baseUrl: BASE_URL,
+        owner,
+        repo,
+        ...(branch && { ref: branch }),
+      },
+    );
+    const content = Buffer.from(response.data.content, 'base64').toString(
+      'utf8',
+    );
+
+    const mediaLinks = [
+      ...content.matchAll(
+        /\[([^\[\]]*)\]\((?!https?:\/\/)(.*?)(\.png|\.jpg|\.jpeg|\.gif|\.webp|\.svg)(.*)\)/gim,
+      ),
+    ].map(match => [match[2], match[3]].join(''));
+
+    const media: Record<string, string> = {};
+
+    for (const href of mediaLinks) {
+      const mimeType = mimeTypeLookup(href);
+
+      if (mimeType) {
+        const contentResponse = await octokit.request(
+          `GET /repos/${owner}/${repo}/contents/${href}`,
+        );
+        media[
+          href
+        ] = `data:${mimeType};base64,${contentResponse.data.content.replaceAll(
```

Comment:
This is the magic. It's creating src urls with the data protocol. I.e. inline image src.